src/Application/GamePathResolver.h

```diff
@@ -0,0 +1,17 @@
+﻿#include <string>
+
+
+#include "Platform/Platform.h"
```

Comment:
Just forward-declare Platform here?

Reply:
i don't like forward-declaring and i think it goes against the google guidelines
we used to do a lot of forward declarations in this project before and ended up having trouble to actually #include in cpp files later (conflicts, order of inclusions mattered, etc) small problems build up over time
while if you always include the full header you are verifying on spot that 1. it works  2. it's self-sufficient and works in any order (as different files include your stuff arbitrary)
i think that's the reasoning behind the google's suggestion to always include 

unless the guidelines changed since

the downside is compile time and more frequent recompilation required
but to address that we might want to look into cmake's precompiled headers

Reply:
Let's discuss on Discord. I disagree with the opinion that forward-declarations are troublesome.